src/Core/SecretsManager/Entities/ApiKey.cs

```diff
@@ -2,22 +2,24 @@
 using Bit.Core.Entities;
 using Bit.Core.Utilities;
 
+#nullable enable
+
 namespace Bit.Core.SecretsManager.Entities;
 
 public class ApiKey : ITableObject<Guid>
 {
     public Guid Id { get; set; }
     public Guid? ServiceAccountId { get; set; }
     [MaxLength(200)]
-    public string Name { get; set; }
+    public string Name { get; set; } = null!;
```

Comment:
It seems weird initializing a non-nullable value type as null. 

Have we thought about using the `required` modifier or initializing as an empty string?

Reply:
It's a pretty common scenario to tell the compiler you know something that it doesn't, in this case that this type is often created through reflection and from data that is guaranteed in another way to be non-null. 

I'm definitely all for using `required` or even constructor parameter which both Dapper and EF will support but unfortunately that caused build errors (mostly in test code) that I felt was out of scope for now. 

I also prefer `null!` to `""` for two reasons. One, the JIT compiler knows to no-op it so it should be faster and two, because it could cover up a problem even more so. You could try and create a `new ApiKey()` and you were supposed to set the name but missed it. Now you might try to do `apiKey.Name.Trim()` and it would work with you not knowing as early that you missed assigning it an actual value. Or since it's not-null it would save into the database correctly. If it was left as null then it would throw an error.

Reply:
Thank you for the detailed explanation!

I prefer `required` but I'm good with moving forward this way to get to enabling this feature project wide.